### PR TITLE
Latejoin Spawn Selection

### PR DESCRIPTION
**Pull request by DaveTheHeadcrab**
You may now select a preference for your latejoin spawn location.
Humans may latespawn at the arrivals shuttle or cryodorms.
Cyborgs may latespawn at the arrivals shuttle or the AI satellite robot storage unit thing.

There was support for latespawning at the gateway as well, but I have decided to leave that out for now as I've had my fill of controversy for a while.

:cl: DaveTheHeadcrab
rscadd: You may now select your latejoin spot in the character setup menu
/:cl:

### CODE REVIEW

File: code/modules/client/preference/preferences_spawnpoints.dm [hunk 1]
```diff
@@ -30,13 +30,15 @@ var/list/spawntypes = list()
 	..()
 	turfs = latejoin
 
+/*
 /datum/spawnpoint/gateway
```
Comment (tigercat2000) [inline code/modules/client/preference/preferences_spawnpoints.dm]: rawr remove the code if you don't want to use it

File: code/modules/client/preference/preferences_mysql.dm [hunk 2]
```diff
@@ -382,6 +385,7 @@
 												socks='[socks]',
 												body_accessory='[body_accessory]',
 												gear='[gearlist]'
```
Comment (tigercat2000) [inline code/modules/client/preference/preferences_mysql.dm]: add a comma

**Comment by Earthdivine** [comment]
Can't think of any reason why we wouldn't allow gateway spawning. Though watch as someone proves me other wise.

:+1:

**Comment by tigercat2000** [comment]
@Earthdivine because there's no way out from the gateway? door's locked

**Comment by DaveTheHeadcrab** [comment]
Because I'd have to unlock the door.
And people would complain about letting everyone jump ship to the gateway instead of staying on the station.
And then the PR would get held up.
And then Dave has to eat more children to stay calm.

**Comment by Spacemanspark** [comment]
:+1: Sure.

**Comment by LittleBigKid2000** [comment]
Clown gets to choose to either spawn in the permabrig, solitary, execution room, or the engine containment area thing.

File: code/modules/client/preference/preferences_spawnpoints.dm (hunk 1 above)
Comment (monster860) [inline code/modules/client/preference/preferences_spawnpoints.dm]: But comments = less diff

**Comment by Earthdivine** [comment]
I forgot the gateway door starts locked. whoops.

**Comment by theColdflame** [comment]
:+1: as it stands, but gateway spawning is dumb

**Comment by Twinmold93** [comment]
The other issue with gateway spawning would be mostly with if the gateway is locked down, anyone who spawns there would, effectively, be stuck. Also, another area we'd have to considered 'protected', which leads to less overall freedoms for antags/could be bad for swarmers spawning and then getting screwed over because someone had just spawned there as well. I'm not against gateway spawning, just need a few things sorted first for that.

**Comment by Sanpd** [comment]
I think this would be handy feature and thus I give my thumbs up for it.

**Comment by TheDZD** [comment]
Make sure this doesn't break the database with the default value.

**Comment by FalseIncarnate** [comment]
Make sure this works with metastation, otherwise :+1:

**Comment by Fox-McCloud** [comment]
Please test this with an actual database.

**Comment by Fox-McCloud** [comment]
Closing this; stale. Submitter has stated he's unlikely to be able to finish this/too busy.

**Pull request closed without merging**